### ISSUE TITLE
[Accessibility] [Toggle] Form control with "switch" role has no associated label

### ISSUE THREAD

**Issue by eduardmarcinco** [carbon-design-system/carbon-components-angular#2696]
## Detailed description

> Describe in detail the issue you're having.

Accessibility checker complains about a violation in toggle component.

> Is this a feature request (new component, new icon), a bug, or a general issue?

Accessibility violation.

> Is this issue related to a specific component?

Toggle.

> What version of the Carbon Design System are you using?

`"carbon-components-angular": "5.14.10"`

## Steps to reproduce the issue

Violation can be seen in storybook.

https://angular.carbondesignsystem.com/iframe.html?args=&id=components-toggle--basic&viewMode=story

![image](https://github.com/carbon-design-system/carbon-components-angular/assets/359643/933caf89-3fed-41e4-ad87-139b0315eee9)
